packages/angular-material/src/layouts/array-layout.renderer.ts

```diff
@@ -0,0 +1,148 @@
+  in the Software without restriction, including without limitation the rights
+  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+  copies of the Software, and to permit persons to whom the Software is
+  furnished to do so, subject to the following conditions:
+
+  The above copyright notice and this permission notice shall be included in
+  all copies or substantial portions of the Software.
+
+  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+  THE SOFTWARE.
+*/
+import { ChangeDetectionStrategy, Component } from '@angular/core';
+import {
+  JsonFormsAngularService,
+  JsonFormsArrayControl
+} from '@jsonforms/angular';
+import {
+  ArrayControlProps,
+  createDefaultValue,
+  findUISchema,
+  isObjectArrayWithNesting,
+  mapDispatchToArrayControlProps,
+  OwnPropsOfRenderer,
+  Paths,
+  RankedTester,
+  rankWith,
+  setReadonly,
+  UISchemaElement,
+  UISchemaTester,
+  unsetReadonly,
+} from '@jsonforms/core';
+
+@Component({
+  selector: 'app-array-layout-renderer',
+  template: `
+    <div fxLayout="column" fxLayoutGap="16px" [fxHide]="hidden">
+      <mat-toolbar>
+        <h2>{{ this.label }}</h2>
+        <span fxFlex></span>
+        <button
+          mat-icon-button
+          [disabled]="!isEnabled()"
+          (click)="add()"
+          attr.aria-label="{{ 'Add to ' + this.label + 'button' }}"
+        >
+          <mat-icon>add</mat-icon>
+        </button>
+      </mat-toolbar>
+      <p *ngIf="noData">No data</p>
+      <div
+        *ngFor="
+          let item of data;
+          let idx = index;
+          trackBy: trackByFn;
+          last as last
+        "
+      >
+        <mat-card>
+          <mat-card-content>
+            <jsonforms-outlet [renderProps]="getProps(idx)"></jsonforms-outlet>
+          </mat-card-content>
+          <mat-card-actions *ngIf="isEnabled()">
+            <button mat-button (click)="remove(idx)" attr.aria-label="Remove">
+              <mat-icon>remove_circle</mat-icon>
+              Remove
+            </button>
+          </mat-card-actions>
+        </mat-card>
+      </div>
+    </div>
+  `,
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class ArrayLayoutRenderer extends JsonFormsArrayControl {
+  noData: boolean;
+  addItem: (path: string, value: any) => () => void;
+  removeItems: (path: string, toDelete: number[]) => () => void;
+  uischemas: {
+    tester: UISchemaTester;
+    uischema: UISchemaElement;
+  }[];
+
+  constructor(private jsonFormAngularService: JsonFormsAngularService) {
```

Comment:
we don't need to make this private as we can access it from the parent

Reply:
If not, it should be accessible via the parent

Reply:
Fixed